Cesium.CodeGen/Ir/Expressions/BinaryOperators/ArithmeticBinaryOperatorExpression.cs

```diff
@@ -93,41 +131,16 @@ public override IType GetExpressionType(IDeclarationScope scope)
         var leftType = Left.GetExpressionType(scope);
         var rightType = Right.GetExpressionType(scope);
 
-        if (leftType is PointerType || rightType is PointerType)
+        switch (leftType, rightType)
         {
-            if (leftType is PointerType)
-            {
-                return leftType;
-            }
+            case (PointerType, not PointerType): return leftType;
+            case (not PointerType, PointerType): return rightType;
+            case (PointerType left, PointerType right):
+                Debug.Assert(left.Base.GetSizeInBytes(scope.ArchitectureSet) == right.Base.GetSizeInBytes(scope.ArchitectureSet));
 
-            return rightType;
+                return scope.CTypeSystem.Int; // ptrdiff_t, must be signed, int is "fine" (nint is not numeric anyway)
```

Comment:
Can we have NativeInt here? Because on 64-bit platforms that would be 64-bit value, but Int is 32bit.

Reply:
This is the same as in #374 - nativeint is not supported in GetCommonNumericType, which is used in some places, so there's more to fix before we can use nativeint here

Reply:
Should be supported already (or if not, then should be easy to implement), so let's do it.

Reply:
Rebased and fixed